Source/WTF/wtf/URLParser.cpp

```diff
@@ -2703,12 +2703,9 @@ bool URLParser::subdomainStartsWithXNDashDash(CodePointIterator<CharacterType> i
 
 bool URLParser::subdomainStartsWithXNDashDash(StringImpl& host)
 {
-    if (host.is8Bit()) {
-        const LChar* begin = host.characters8();
-        return subdomainStartsWithXNDashDash(CodePointIterator<LChar>(begin, begin + host.length()));
-    }
-    const UChar* begin = host.characters16();
-    return subdomainStartsWithXNDashDash(CodePointIterator<UChar>(begin, begin + host.length()));
+    if (host.is8Bit())
+        return subdomainStartsWithXNDashDash(CodePointIterator<LChar>(host.span8()));
```

Comment:
Can we rearrange things so we don’t have to write the class name `CodePointIterator` out here? Maybe if we add a deduction guide we can implicitly convert in most places like this.